core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1295,16 +1295,23 @@ private RecordCursorFactory generateOrderBy(RecordCursorFactory recordCursorFact
                 // if first column index is the same as timestamp of underling record cursor factory
                 // we could have two possibilities:
                 // 1. if we only have one column to order by - the cursor would already be ordered
-                //    by timestamp; we have nothing to do
+                //    by timestamp (either ASC or DESC); we have nothing to do
                 // 2. metadata of the new cursor will have timestamp
 
                 RecordMetadata orderedMetadata;
                 if (metadata.getTimestampIndex() != -1) {
                     CharSequence column = columnNames.getQuick(0);
                     int index = metadata.getColumnIndexQuiet(column);
                     if (index == metadata.getTimestampIndex()) {
-                        if (size == 1 && orderBy.get(column) == QueryModel.ORDER_DIRECTION_ASCENDING) {
-                            return recordCursorFactory;
+                        if (size == 1) {
+                            if (orderBy.get(column) == QueryModel.ORDER_DIRECTION_ASCENDING) {
+                                return recordCursorFactory;//DataFrameRecordCursorFactory
```

Comment:
This piece reverses order of Fwd data frame/row cursor (if needed) and removes the need to sort data.

Unfortunately order by is not available when/where record cursor factory is created (subquery) so this seemed like the only feasible option (for now).
Suggestions welcome !